### PR TITLE
Return a function instead of calling it (in safe_eval)

### PR DESCRIPTION
**Pull request by vloncar**
If quantizer is a function (like `binary_tanh` or `hard_sigmoid`, used as activations), `safe_eval` would try to make an instance of it and fail. This was due to the change introduced in #12. We should check if quantizer is class to be instantiated before being called or a function ready to be called.

### CODE REVIEW

File: qkeras/safe_eval.py [hunk 1]
```diff
@@ -83,4 +83,7 @@ def safe_eval(eval_str, op_dict, *params, **kwparams):  # pylint: disable=invali
   if len(function_split) == 2 or args or kwargs:
     return quantizer(*args, **kwargs)
   else:
-    return quantizer()
+    if isinstance(quantizer, type):
```
Comment (zhuangh) [inline qkeras/safe_eval.py]: please add comment here, something like "check the quantizer is a class"

**Comment by vloncar** [comment]
> x = QConv2D(64, (3, 3), strides=(2,2), kernel_quantizer=binary_tanh, bias_quantizer=quantized_bits(4,0,1), name="conv2d_1_m", activation="hard_sigmoid")(x)
> x = QActivation("hard_sigmoid")(x)

I tried, but if i understand correctly, this is not how quantizers are intended to be used. The first `hard_sigmoid` is Keras's one, and the second one is QKeras's one and we never use this combination in our models. It would be useful to be able to use QKeras activations without using QActivation layer, but that is a separate issue (which we can look into).

**Comment by zhuangh** [comment]
Since the PR is for "If quantizer is a function (like binary_tanh or hard_sigmoid, used as activations), ...", I just want to try whether it can be used in activation inside a qkeras layers and QActivation layer directly.

**Comment by vloncar** [comment]
Using `QConv2d(..., activation='some_qkeras_activation'...)` won't work (but we are interested in having that). Using `QActivation('some_qkeras_activation'..)` works for all quantizers. Combining the two doesn't make sense, i.e., anything before `QAactivation` should be `activation=None` or `activation='linear'`.

**Comment by vloncar** [comment]
Is there anything else I should do for this PR?

**Comment by zhuangh** [comment]
sorry about the late. no for now. I am waiting for feedback from another googler (required by the internal flow).

**Pull request merged**